### PR TITLE
feat(ffi): add a client example using Avalonia

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**
1. Add example RDP client using Avalonia
2. Generate extra bindings necessary 

Demo:

https://github.com/Devolutions/IronRDP/assets/139169536/548cbad0-6d6f-4907-8784-b6b59300eb53

Note: As this is an example for demonstration purpose, many things are not implemented.
Currently only simple keyboard and cursor move/click events are implemented.

### CODE REVIEW

File: ffi/src/connector/state.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ pub mod ffi {
     }
 
     impl ClientConnectorState {
-        pub fn get_type(&self) -> Result<ClientConnectorStateType, Box<IronRdpError>> {
+        pub fn get_enum_type(&self) -> Result<ClientConnectorStateType, Box<IronRdpError>> {
```
Comment (irvingoujAtDevolution) [inline ffi/src/connector/state.rs:30]: avoid naming conflict with C# GetType()

File: ffi/dotnet/Devolutions.IronRdp.AvaloniaExample/App.axaml.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using Avalonia.Controls.ApplicationLifetimes;
+using Avalonia.Markup.Xaml;
+
+namespace Devolutions.IronRdp.AvaloniaExample;
+
+public partial class App : Application
+{
+    public override void Initialize()
+    {
+        Console.WriteLine("App Initialize");
```
Comment (CBenoit) [inline ffi/dotnet/Devolutions.IronRdp.AvaloniaExample/App.axaml.cs:12]: question: We probably don’t really need such outputs in the terminal

File: ffi/dotnet/Devolutions.IronRdp/Generated/Char.cs
Comment (CBenoit) [inline ffi/dotnet/Devolutions.IronRdp/Generated/Char.cs:1]: issue: This is overkill. Use u32 instead.

File: ffi/dotnet/Devolutions.IronRdp/src/Framed.cs [hunk 3]
```diff
@@ -20,19 +21,21 @@ public Framed(S stream)
...
             {
+                Console.WriteLine("ReadPdu: ReadExact, pduInfo.GetLength()=" + pduInfo.GetLength());
                 var frame = await this.ReadExact(pduInfo.GetLength());
                 var action = pduInfo.GetAction();
                 return (action, frame);
             }
             else
             {
                 var len = await this.Read();
+                Console.WriteLine("ReadPdu: Read, len=" + len);
```
Comment (CBenoit) [inline ffi/dotnet/Devolutions.IronRdp/src/Framed.cs:39]: issue: The output may be too verbose for a basic terminal output. This is also part of the library, so all of this should be removed.

File: ffi/src/connector/config.rs [hunk 4]
```diff
@@ -120,7 +117,9 @@ pub mod ffi {
             self.performance_flags = Some(performance_flags.0);
         }
 
-        // TODO: set bitmap
+        pub fn set_bitmap(&mut self, bitmap: &BitmapConfig) {
```
Comment (CBenoit) [inline ffi/src/connector/config.rs:120]: naming:
```suggestion
        pub fn set_bitmap_config(&mut self, bitmap: &BitmapConfig) {
```

File: ffi/src/input.rs [hunk 5]
```diff
@@ -1,11 +1,131 @@
...
+
+    #[diplomat::opaque]
+    pub struct MousePosition(pub ironrdp::input::MousePosition);
+
+    impl MousePosition {
+        pub fn new(x: u16, y: u16) -> Box<MousePosition> {
+            Box::new(MousePosition(ironrdp::input::MousePosition { x, y }))
+        }
+
+        pub fn as_operation(&self) -> Box<Operation> {
```
Comment (CBenoit) [inline ffi/src/input.rs:41]: naming: `as_move_operation`
```suggestion
        pub fn as_move_operation(&self) -> Box<Operation> {
```

File: ffi/src/input.rs [hunk 6]
```diff
@@ -1,11 +1,131 @@
...
+
+        pub fn as_operation_unicode_key_pressed(&self) -> Box<Operation> {
+            let operation = ironrdp::input::Operation::UnicodeKeyPressed(self.0);
+            Box::new(Operation(operation))
+        }
+
+        pub fn as_operation_unicode_key_released(&self) -> Box<Operation> {
+            let operation = ironrdp::input::Operation::UnicodeKeyReleased(self.0);
+            Box::new(Operation(operation))
+        }
```
Comment (CBenoit) [inline ffi/src/input.rs:129]: suggestion: Use `u32` and [`char::from_u32`](https://doc.rust-lang.org/std/primitive.char.html#method.from_u32) instead

**Comment by CBenoit** [comment]
Nice job! Do you have a demonstration video?

**Comment by irvingoujAtDevolution** [comment]
> Nice job! Do you have a demonstration video?

Do you mean a demo video for running the example? 
https://github.com/Devolutions/IronRDP/assets/139169536/548cbad0-6d6f-4907-8784-b6b59300eb53

File: ffi/dotnet/Devolutions.IronRdp.AvaloniaExample/App.axaml.cs (hunk 2 above)
Comment (irvingoujAtDevolution) [reply ffi/dotnet/Devolutions.IronRdp.AvaloniaExample/App.axaml.cs]: updated

File: ffi/dotnet/Devolutions.IronRdp/Generated/Char.cs
Comment (irvingoujAtDevolution) [reply ffi/dotnet/Devolutions.IronRdp/Generated/Char.cs:1]: updated

File: ffi/dotnet/Devolutions.IronRdp/src/Framed.cs (hunk 3 above)
Comment (irvingoujAtDevolution) [reply ffi/dotnet/Devolutions.IronRdp/src/Framed.cs]: updated

File: ffi/src/input.rs (hunk 6 above)
Comment (irvingoujAtDevolution) [reply ffi/src/input.rs]: updated

**Review by CBenoit — APPROVED** [review]
Great job!

**Pull request merged**